gui/src/components/find/FindWidget.tsx

```diff
@@ -159,20 +163,29 @@ export const useFindWidget = (searchRef: RefObject<HTMLDivElement>) => {
     });
 
     resizeObserver.observe(searchRef.current);
+    if (headerRef.current) {
+      resizeObserver.observe(headerRef.current);
+    }
     return () => {
       if (timeoutId) {
         clearTimeout(timeoutId);
       }
-      if (searchRef.current) resizeObserver.unobserve(searchRef.current);
+      if (searchRef.current) {
+        resizeObserver.unobserve(searchRef.current);
+      }
+      if (headerRef.current) {
+        resizeObserver.unobserve(headerRef.current);
+      }
     };
-  }, [searchRef.current]);
+  }, [searchRef, headerRef]);
 
   // Main function for finding matches and generating highlight overlays
   const refreshSearch = useCallback(
```

Comment:
The `refreshSearch` is large and complicated as is and `useFindWidget` is 500 lines long. How about moving it this function of `FindWidget` body? 

```suggestion
  const refreshSearch = useCallback(someNewFunction({
  headerRef,
      searchRef,
      debouncedInput,
      scrollToMatch,
      caseSensitive,
      useRegex,
  })
```